### PR TITLE
Throw debugmsg when setting charges for items that cannot have charges

### PR DESCRIPTION
**Pull request by RenechCDDA**
#### Summary
Bugfixes "Butchery properly gives a scaling amount of butchery refuse, instead of just 1"

#### Purpose of change
@GuardianDll raised a question in the development discord about butchery only giving 1 butchery refuse. Some checks later, I found out that constructor naively sets charges regardless of whether the type is allowed to have them or not. This results in an item with a `charges` defined, but that is handled by our code as not being `count_by_charges()`, thus ignoring the value.

#### Describe the solution
Only set charges if type is allowed to have charges
Debugmsg if type is not allowed to have charges

Fix existing butchery call

#### Additional context
"Charges are the devil" -kevingranade

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -312,7 +312,11 @@ item::item( const itype *type, time_point turn, int qty ) : type( type ), bday(
     }
 
     if( qty >= 0 ) {
-        charges = qty;
+        if( count_by_charges() ) {
```
Comment (mqrause) [inline src/item.cpp:315]: Using `type->can_have_charges()` here might be safer. But tbh I'm not sure if there's cases where this is relevant. (Yes, having charges and being count_by_charges aren't necessarily the same thing.)

**Comment by RenechCDDA** [comment]
Oh yeah, the tests absolutely explode.

We have deeper problems than just butchery, don't we?

**Comment by mqrause** [comment]
Chances are 99%+ of those errors set charges to 1. Because any value bigger than that should trigger charge removal after a save/load cycle, causing stuff to fall out once you interact with it. And I'd expect that would have been noticed already.

**Comment by RenechCDDA** [comment]
Yeah that's what I'm seeing when running the tests locally, just lots of calls to the constructor with irrelevant quantities. I think I'm just going to paper over it by only throwing error on qty > 1

**Comment by mqrause** [comment]
Chances are also good this is coming from limited sources like item group creation.

**Comment by RenechCDDA** [comment]
Assuming this pass succeeds on GH I am going to consider removing the `if qty > 1` requirement and eliminating the rest of the malcontents.

**Comment by PatrikLundell** [comment]
If there are too many complaints I'd suggest checking for >1 as suggested and then initiate a series of cleanup PRs, ending with the removal of the incorrectness hiding check. Especially in the light of the recent "too large" PR rules.

File: src/mattack_actors.cpp [hunk 2]
```diff
@@ -1273,7 +1273,6 @@ void gun_actor::shoot( monster &z, const tripoint &target, const gun_mode_id &mo
     }
 
     tmp.set_wielded_item( gun );
-    tmp.i_add( item( "UPS_off", calendar::turn, 1000 ) );
```
Comment (akrieger) [inline src/mattack_actors.cpp:1276]: Haha wtf, when a monster shoots we gin up a fake npc to actually do the shooting?

**Comment by akrieger** [comment]
IDK seems green.

**Pull request merged**